pkg/customresourcestate/registry_factory.go

```diff
@@ -99,6 +99,9 @@ func fullName(resource Resource, f Generator) string {
 		parts = append(parts, resource.GetMetricNamePrefix())
 	}
 	parts = append(parts, f.Name)
+	if resource.wasResolved {
+		parts = append(parts, resource.GroupVersionKind.String())
```

Comment:
This will probably need to be documented. We are essentially appending the `g_v_k` (in that format) of the object to the `kube_customresource_name` metric format here, otherwise we'd have duplicate metric names.

Reply:
I thought we moved the GVK into the labels.. are we re-embeding the GVK in the metric name again? I'm not quite sure I follow the issue.

Reply:
Done.

Reply:
GitHub comments are wierd, I saw your comment after commenting a moment ago. I think this skipped my mind, I've fixed this.

Reply:
The documentation is still outdated then, I think.

Reply:
Yeah, updated.